lib/google_search_data_viewer_web/views/keyword_view.ex

```diff
@@ -4,4 +4,19 @@ defmodule GoogleSearchDataViewerWeb.KeywordView do
   def format_date_time(datetime) do
     Calendar.strftime(datetime, "%d.%m.%y %H:%M:%S")
   end
+
+  def total_urls(search_results) do
+    search_results
+    |> Enum.map(fn {_, search_result_urls} -> Enum.count(search_result_urls) end)
+    |> Enum.sum()
+  end
+
+  def total_top_adword_urls(search_results),
+    do: Enum.count(Map.get(search_results, :top_adwords, []))
+
+  def total_adword_urls(search_results),
+    do: Enum.count(Map.get(search_results, :adwords, []))
+
+  def total_non_adword_urls(search_results),
+    do: Enum.count(Map.get(search_results, :non_adwords, []))
```

Comment:
How about having only one helper method that can be used with all search result types?
```elixir
def count_search_result(search_results, type) do
  search_results
  |> Map.get(type, []))
  |> Enum.count()
end
```

Reply:
Yup works, just required to input additional arguments to the function but when in the future a different type of URL is required then this would not need to be modified. 

I am calling this method multiple times throughout the template, wonder if there's a better way to store the initial result.

Reply:
The one way I can think of is to build the value in the controller and pass it to the view as assigns. But for this case, IMO the data is not large, so it's fine to call multiple times.